eventbuffer.go

```diff
@@ -10,15 +10,21 @@ import (
 
 // eventBuffer is a buffer for storing outstanding events that need to be processed by the node.
 // It contains a separate list for each module.
-type eventBuffer map[t.ModuleID]*events.EventList
+type eventBuffer struct {
+	buffers     map[t.ModuleID]*events.EventList
+	totalEvents int
+}
 
 // newEventBuffer allocates and returns a pointer to a new eventBuffer object.
-func newEventBuffer(modules modules.Modules) eventBuffer {
+func newEventBuffer(modules modules.Modules) *eventBuffer {
```

Comment:
The `eventBuffer` structure doesn't seem to have grown a lot, why returning it by reference?

Reply:
It turned out that the `AddEvents` method must have a pointer receiver, otherwise a copy of the event counter would be incremented instead of the counter itself (it cost me some debugging time to figure that out... 🙄 ), so I just switched everything to pointers.

But indeed that's not necessary, so switching back to value, only keeping that method receiver as a pointer.